src/Eshopworld.Core/Events/IEventConsumer.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Eshopworld.Core.Events
+{
+    /// <summary>
+    /// Contract that provides a way to send events to the event streaming bus
+    /// </summary>
+    public interface IEventConsumer
+    {
+        /// <summary>
+        /// Start consuming events from the configured topics
+        /// </summary>
+        /// <param name="startFromBeginningWhenNoPointer">If no consumer's group offset is stored on Kafka and this is set to true the consumer will consume all existing events in the topic</param>
+        /// <param name="cancelToken">Token that can be passed to cancel the operation.</param>
+        Task StartConsumingAsync(bool startFromBeginningWhenNoPointer = false, CancellationToken cancelToken = default);
+
+        /// <summary>
+        /// Start consuming events from the configured topics from a certain date & time regardless of the stored consume's group offset
+        /// </summary>
+        /// <param name="start">The date & time to start consuming from.</param>
+        /// <param name="cancelToken">Token that can be passed to cancel the operation.</param>
+        Task StartConsumingFromDateTimeAsync(DateTimeOffset start, CancellationToken cancelToken = default);
```

Comment:
Are the pointers already date times?

Reply:
Offsets are ints but infrastructure likes Kafka allows us to lookup the offset based on a date & time. Date & time makes it more understandable to use from a human point of view, use case would be consuming all transactional data from a certain date & time for rebuilding a dwh for reporting purposes

Reply:
Does it make sense then to have two interfaces here then? one for data time offset and another for the int offset?

Reply:
The int offset would be per partition, meaning to have to specify the topic and per partition the offset you want to start from. I'm not sure if that's relevant for any future use case

Reply:
yeah ok, we can leave that one for now